perfdash/www/script.js

```diff
@@ -19,103 +19,104 @@ var app = angular.module('PerfDashApp', ['ngMaterial', 'chart.js']);
 PerfDashApp.prototype.refresh = function() {
-    this.http.get("api")
-    .success(function(data) {
-        this.testNames = Object.keys(data);
-        this.testName = this.testNames[0];
-        this.allData = data;
-        this.testNameChanged();
-        this.labels = this.getLabels();
-        this.resources = this.getResources();
-        this.verbs = this.getVerbs();
-    }.bind(this))
+    this.http.get("result")
+            .success(function(data) {
+                this.testNames = Object.keys(data);
+                this.testName = this.testNames[0];
+                this.allData = data;
+                this.testNameChanged();
+            }.bind(this))
     .error(function(data) {
-        console.log("error fetching api");
+        console.log("error fetching result");
         console.log(data);
     });
 };
 
-// Update the data to graph, using the selected resource and verb
-PerfDashApp.prototype.resourceChanged = function() {
-    this.seriesData = [
-        this.getStream(this.getData(this.selectedResource, this.selectedVerb), "Perc99"),
-        this.getStream(this.getData(this.selectedResource, this.selectedVerb), "Perc90"),
-        this.getStream(this.getData(this.selectedResource, this.selectedVerb), "Perc50")
-    ];
+// Update the data to graph, using selected labels
+PerfDashApp.prototype.labelChanged = function() {
+    this.seriesData = [];
+    this.series = [];
+    result = this.getData(this.selectedLabels);
+    if (result.length <= 0) {
+        return;
+    }
+    // All the unit should be the same
+    this.options= {scaleLabel: "<%=value%> "+result[0].unit};
+    angular.forEach(result[0].data, function(value, name) {
+        this.seriesData.push(this.getStream(result, name));
+        this.series.push(name);
+    }, this);
 };
 
 // Update the data to graph, using the selected testName
 PerfDashApp.prototype.testNameChanged = function() {
-    this.data = this.allData[this.testName];
-    this.resourceChanged();
+    this.data = this.allData[this.testName].builds;
+    this.job = this.allData[this.testName].job;
+    this.builds = this.getBuilds();
+    this.labels = this.getLabels();
+    this.labelChanged();
 };
 
-// Get the set of all resources (e.g. 'pods') in the data set
-PerfDashApp.prototype.getResources = function() {
-    var set = {};
-    angular.forEach(this.data, function(value, key) {
-        angular.forEach(value, function(v, k) {
-            set[k] = true;
-        })
-    });
-    var result = [];
-    angular.forEach(set, function(value, key) {
-        result.push(key);
-    })
-    return result;
+// Get all of the builds for the data set (e.g. build numbers)
+PerfDashApp.prototype.getBuilds = function() {
+    return Object.keys(this.data)
 };
 
-// Get the set of all verbs (e.g. 'GET') in the data set
-PerfDashApp.prototype.getVerbs = function() {
+// Get the set of all labels (e.g. 'resources', 'verbs') in the data set
+PerfDashApp.prototype.getLabels = function() {
```

Comment:
`getLabels` method that returns nothing? Perhaps change name to something that actually explains what this does? :)

Applies to getData too